### ISSUE TITLE
antsApplyTransforms with c3d-converted affine:  itk::ERROR: TxtTransformIOTemplate(0x2912350): Could not create an instance of MatrixOffsetTransformBase_double_3_3

### ISSUE THREAD

**Issue by sgiavasis** [ANTsX/ANTs#84]
Hello,

I am trying to run antsApplyTransforms where one of the transforms is a c3d-converted affine from FSL FLIRT (.mat -> .txt). When I run the command I get this error (the final --transform, affine.txt is the one in question):

antsApplyTransforms --default-value 0 --dimensionality 4 --input /home/likewise-open/CHILDMIND/steven.giavasis/run/035/035_bug/w2/resting_preproc_2014113/_scan_rest_1_rest/_csf_threshold_0.98/_gm_threshold_0.7/_wm_threshold_0.98/_compcor_ncomponents_5_selector_pc10.linear1.wm0.global0.motion1.quadratic0.gm0.compcor1.csf0/_bandpass_freqs_0.009.0.1/frequency_filter_0/bandpassed_demeaned_filtered.nii.gz --interpolation Gaussian --output bandpassed_demeaned_filtered_antswarp.nii.gz --reference-image /home/likewise-open/CHILDMIND/steven.giavasis/run/035/035_bug/w2/resting_preproc_2014113/anat_preproc_0/anat_brain_only/mprage_resample_calc.nii.gz --transform /home/likewise-open/CHILDMIND/steven.giavasis/run/035/035_bug/w2/resting_preproc_2014113/anat_mni_ants_register_0/calculate_ants_warp/transform2Warp.nii.gz --transform /home/likewise-open/CHILDMIND/steven.giavasis/run/035/035_bug/w2/resting_preproc_2014113/anat_mni_ants_register_0/calculate_ants_warp/transform1Affine.mat --transform /home/likewise-open/CHILDMIND/steven.giavasis/run/035/035_bug/w2/resting_preproc_2014113/fsl_to_itk_func_mni_4d_0/_scan_rest_1_rest/fsl_reg_2_itk/affine.txt

Using double precision for computations.
Input scalar image: /home/likewise-open/CHILDMIND/steven.giavasis/run/035/035_bug/w2/resting_preproc_2014113/_scan_rest_1_rest/_csf_threshold_0.98/_gm_threshold_0.7/_wm_threshold_0.98/_compcor_ncomponents_5_selector_pc10.linear1.wm0.global0.motion1.quadratic0.gm0.compcor1.csf0/_bandpass_freqs_0.009.0.1/frequency_filter_0/bandpassed_demeaned_filtered.nii.gz
Reference image: /home/likewise-open/CHILDMIND/steven.giavasis/run/035/035_bug/w2/resting_preproc_2014113/anat_preproc_0/anat_brain_only/mprage_resample_calc.nii.gz
Transform reader for /home/likewise-open/CHILDMIND/steven.giavasis/run/035/035_bug/w2/resting_preproc_2014113/fsl_to_itk_func_mni_4d_0/_scan_rest_1_rest/fsl_reg_2_itk/affine.txt caught an ITK exception:

itk::ExceptionObject (0x2912c10)
Location: "unknown" 
File: /home/likewise-open/CHILDMIND/steven.giavasis/antsbin_NEW/ITKv4/Modules/IO/TransformBase/include/itkTransformIOBase.hxx
Line: 70
Description: itk::ERROR: TxtTransformIOTemplate(0x2912350): Could not create an instance of MatrixOffsetTransformBase_double_3_3
The usual cause of this error is not registering the transform with TransformFactory
Currently registered Transforms: 
    "AffineTransform_double_2_2"
    "AffineTransform_double_3_3"
    "AffineTransform_double_4_4"
    "AffineTransform_double_5_5"
    "AffineTransform_double_6_6"
    "AffineTransform_double_7_7"
    "AffineTransform_double_8_8"
    "AffineTransform_double_9_9"
    "AffineTransform_float_2_2"
    "AffineTransform_float_3_3"
    "AffineTransform_float_4_4"
    "AffineTransform_float_5_5"
    "AffineTransform_float_6_6"
    "AffineTransform_float_7_7"
    "AffineTransform_float_8_8"
    "AffineTransform_float_9_9"
    "BSplineDeformableTransform_double_2_2"
    "BSplineDeformableTransform_double_3_3"
    "BSplineDeformableTransform_float_2_2"
    "BSplineDeformableTransform_float_3_3"
    "BSplineSmoothingOnUpdateDisplacementFieldTransform_double_2_2"
    "BSplineSmoothingOnUpdateDisplacementFieldTransform_double_3_3"
    "BSplineSmoothingOnUpdateDisplacementFieldTransform_float_2_2"
    "BSplineSmoothingOnUpdateDisplacementFieldTransform_float_3_3"
    "BSplineTransform_double_2_2"
    "BSplineTransform_double_3_3"
    "BSplineTransform_float_2_2"
    "BSplineTransform_float_3_3"
    "CenteredAffineTransform_double_2_2"
    "CenteredAffineTransform_double_3_3"
    "CenteredAffineTransform_float_2_2"
    "CenteredAffineTransform_float_3_3"
    "CenteredEuler3DTransform_double_3_3"
    "CenteredEuler3DTransform_float_3_3"
    "CenteredRigid2DTransform_double_2_2"
    "CenteredRigid2DTransform_float_2_2"
    "CenteredSimilarity2DTransform_double_2_2"
    "CenteredSimilarity2DTransform_float_2_2"
    "CompositeTransform_double_2_2"
    "CompositeTransform_double_3_3"
    "CompositeTransform_double_4_4"
    "CompositeTransform_double_5_5"
    "CompositeTransform_double_6_6"
    "CompositeTransform_double_7_7"
    "CompositeTransform_double_8_8"
    "CompositeTransform_double_9_9"
    "CompositeTransform_float_2_2"
    "CompositeTransform_float_3_3"
    "CompositeTransform_float_4_4"
    "CompositeTransform_float_5_5"
    "CompositeTransform_float_6_6"
    "CompositeTransform_float_7_7"
    "CompositeTransform_float_8_8"
    "CompositeTransform_float_9_9"
    "ConstantVelocityFieldTransform_double_2_2"
    "ConstantVelocityFieldTransform_double_3_3"
    "ConstantVelocityFieldTransform_float_2_2"
    "ConstantVelocityFieldTransform_float_3_3"
    "DisplacementFieldTransform_double_2_2"
    "DisplacementFieldTransform_double_3_3"
    "DisplacementFieldTransform_float_2_2"
    "DisplacementFieldTransform_float_3_3"
    "Euler2DTransform_double_2_2"
    "Euler2DTransform_float_2_2"
    "Euler3DTransform_double_3_3"
    "Euler3DTransform_float_3_3"
    "FixedCenterOfRotationAffineTransform_double_3_3"
    "FixedCenterOfRotationAffineTransform_float_3_3"
    "GaussianExponentialDiffeomorphicTransform_double_2_2"
    "GaussianExponentialDiffeomorphicTransform_double_3_3"
    "GaussianExponentialDiffeomorphicTransform_float_2_2"
    "GaussianExponentialDiffeomorphicTransform_float_3_3"
    "GaussianSmoothingOnUpdateDisplacementFieldTransform_double_2_2"
    "GaussianSmoothingOnUpdateDisplacementFieldTransform_double_3_3"
    "GaussianSmoothingOnUpdateDisplacementFieldTransform_float_2_2"
    "GaussianSmoothingOnUpdateDisplacementFieldTransform_float_3_3"
    "GaussianSmoothingOnUpdateTimeVaryingVelocityFieldTransform_double_2_2"
    "GaussianSmoothingOnUpdateTimeVaryingVelocityFieldTransform_double_3_3"
    "GaussianSmoothingOnUpdateTimeVaryingVelocityFieldTransform_float_2_2"
    "GaussianSmoothingOnUpdateTimeVaryingVelocityFieldTransform_float_3_3"
    "IdentityTransform_double_2_2"
    "IdentityTransform_double_3_3"
    "IdentityTransform_double_4_4"
    "IdentityTransform_double_5_5"
    "IdentityTransform_double_6_6"
    "IdentityTransform_double_7_7"
    "IdentityTransform_double_8_8"
    "IdentityTransform_double_9_9"
    "IdentityTransform_float_2_2"
    "IdentityTransform_float_3_3"
    "IdentityTransform_float_4_4"
    "IdentityTransform_float_5_5"
    "IdentityTransform_float_6_6"
    "IdentityTransform_float_7_7"
    "IdentityTransform_float_8_8"
    "IdentityTransform_float_9_9"
    "MatrixOffsetTransformBase_double_4_4"
    "MatrixOffsetTransformBase_double_4_4"
    "QuaternionRigidTransform_double_3_3"
    "QuaternionRigidTransform_float_3_3"
    "Rigid2DTransform_double_2_2"
    "Rigid2DTransform_float_2_2"
    "Rigid3DPerspectiveTransform_double_3_2"
    "Rigid3DPerspectiveTransform_float_3_2"
    "Rigid3DTransform_double_3_3"
    "Rigid3DTransform_float_3_3"
    "ScalableAffineTransform_double_3_3"
    "ScalableAffineTransform_float_3_3"
    "ScaleLogarithmicTransform_double_3_3"
    "ScaleLogarithmicTransform_float_3_3"
    "ScaleSkewVersor3DTransform_double_3_3"
    "ScaleSkewVersor3DTransform_float_3_3"
    "ScaleTransform_double_2_2"
    "ScaleTransform_double_3_3"
    "ScaleTransform_double_4_4"
    "ScaleTransform_float_2_2"
    "ScaleTransform_float_3_3"
    "ScaleTransform_float_4_4"
    "ScaleVersor3DTransform_double_3_3"
    "ScaleVersor3DTransform_float_3_3"
    "Similarity2DTransform_double_2_2"
    "Similarity2DTransform_float_2_2"
    "Similarity3DTransform_double_3_3"
    "Similarity3DTransform_float_3_3"
    "TimeVaryingBSplineVelocityFieldTransform_double_2_2"
    "TimeVaryingBSplineVelocityFieldTransform_double_3_3"
    "TimeVaryingBSplineVelocityFieldTransform_float_2_2"
    "TimeVaryingBSplineVelocityFieldTransform_float_3_3"
    "TimeVaryingVelocityFieldTransform_double_2_2"
    "TimeVaryingVelocityFieldTransform_double_3_3"
    "TimeVaryingVelocityFieldTransform_float_2_2"
    "TimeVaryingVelocityFieldTransform_float_3_3"
    "TranslationTransform_double_2_2"
    "TranslationTransform_double_3_3"
    "TranslationTransform_double_4_4"
    "TranslationTransform_double_5_5"
    "TranslationTransform_double_6_6"
    "TranslationTransform_double_7_7"
    "TranslationTransform_double_8_8"
    "TranslationTransform_double_9_9"
    "TranslationTransform_float_2_2"
    "TranslationTransform_float_3_3"
    "TranslationTransform_float_4_4"
    "TranslationTransform_float_5_5"
    "TranslationTransform_float_6_6"
    "TranslationTransform_float_7_7"
    "TranslationTransform_float_8_8"
    "TranslationTransform_float_9_9"
    "VelocityFieldTransform_double_2_2"
    "VelocityFieldTransform_double_3_3"
    "VelocityFieldTransform_float_2_2"
    "VelocityFieldTransform_float_3_3"
    "VersorRigid3DTransform_double_3_3"
    "VersorRigid3DTransform_float_3_3"
    "VersorTransform_double_3_3"
    "VersorTransform_float_3_3"

Can't read initial transform /home/likewise-open/CHILDMIND/steven.giavasis/run/035/035_bug/w2/resting_preproc_2014113/fsl_to_itk_func_mni_4d_0/_scan_rest_1_rest/fsl_reg_2_itk/affine.txt

The file that the c3d_affine_tool created, affine.txt, contains this:
# Insight Transform File V1.0
# Transform 0

Transform: MatrixOffsetTransformBase_double_3_3
Parameters: 0.999937 0.00118774 -0.0111388 -0.00188052 0.99805 -0.0623922 0.011043 0.0624092 0.99799 -1.25486 -30.7644 -31.7582
FixedParameters: 0 0 0

So c3d_affine_tool is creating a type of transform which is not included in that list which the antsApplyTransforms error message provided. At first we thought this may be a c3d_affine_tool version or ITK-SNAP version issue, but even with the nightly build version of c3d this still happens.

Any help would be appreciated. Thank you!

**Comment by tina-ebi** [comment]
Hi There,
I would like to register two T1 structural Images. First, I manually registered them by ITK-snap. But when I want to register them by ANTs, I get an error (I have attached pictures). 
any help would be appreciated.

![error](https://user-images.githubusercontent.com/62328344/79108577-e4286480-7d8b-11ea-8750-678527b97c1f.png)
![myscript](https://user-images.githubusercontent.com/62328344/79108582-e5f22800-7d8b-11ea-9d59-62aaf18c0545.png)
